c/driver/postgresql/connection.cc

```diff
@@ -1025,6 +1025,18 @@ AdbcStatusCode PostgresConnection::SetOption(const char* key, const char* value,
     }
     return ADBC_STATUS_OK;
   }
+
+  if (std::strcmp(key, "TIME ZONE") == 0) {
+    std::string query = std::string("SET TIME ZONE '") + std::string(value) + "'";
```

Comment:
Could also set / unset this in the lifecycle of the statement

Reply:
If we do want an option for this, it should be something like `adbc.postgresql.time_zone` based on other drivers

But if we just need it for ingestion, maybe we could do everything inside a transaction, and set/restore the timezone within the transaction?

Reply:
Yea either of those are viable; I think better to start with the latter, though I'm not sure where to manage that lifecycle. Is the `BindStream` struct in statement.cc supposed to be a suitable home for managing a transaction? The type introspection happens there, but I don't know that is the best place to put BEGIN/COMMIT statements.

Reply:
Hmm, I think for this case it might be, but we probably are going to need some more intensive refactoring at some point